File: ConexaoVagas API/Domains/HabilidadeVaga.cs
using System;
using System.Collections.Generic;

namespace ConexaoVagasAPI.Domains
{
    public partial class HabilidadeVaga
    {
        public int IdHabilidadeVaga { get; set; }
        public int IdHabilidade { get; set; }
        public int IdVaga { get; set; }

        public Habilidade IdHabilidadeNavigation { get; set; }
        public Vaga IdVagaNavigation { get; set; }
    }
}

File: ConexaoVagas Front/src/models/habilidadeVaga.ts
import { Habilidade } from "./habilidade";
import { Vaga } from "./vaga";

export class HabilidadeVaga {
    idHabilidadeVaga?: number;
    idHabilidade?: number;
    idVaga?: number;
    idHabilidadeNavigation?: Habilidade = undefined;
    idVagaNavigation?: Vaga = undefined;
}